CPS scope selector changes (#7815)
elastic/docs-content: explore-analyze/cross-project-search/cross-project-search-manage-scope.md

The scope selector lists all linked projects, narrowed by any active [tag filters](#cps-picker-tag-filters). The origin project always appears first, labeled **This project**. Each project row shows the project name, project type icon, and any assigned [tags](/explore-analyze/cross-project-search/cross-project-search-tags.md).

From the scope selector, you can:

* [Toggle individual projects on or off](#cps-picker-include-exclude) to include or exclude them from searches.
* [Filter the project list by tags](#cps-picker-tag-filters) to find and select projects based on metadata or custom tags.

The footer displays the count of included and excluded projects.

The scope selector is not editable in every app. Some apps display it as **read-only**, meaning the app uses the space default scope but you cannot change it. In read-only mode, filter badges are visible but you cannot create, edit, or remove filters. Other apps show it as **unavailable**, meaning the app searches only the current project. Refer to [{{cps-cap}} availability by app](#cps-availability) for details.

> When the current selection matches the [space default](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md#cps-default-search-scope), the scope selector displays a **Using space defaults** indicator. When you change the selection, the indicator disappears. To restore the space default at any time, select **Revert to space defaults** in the scope selector.

Author: ```suggestion
When the current selection matches the [space default](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md#cps-default-search-scope), the scope selector displays a **Using space defaults** indicator. When you change the selection, the indicator disappears. To restore the space default at any time, select **Revert to space defaults** in the scope selector. Changes you make in the project header or on a resource do not change the space default.
```

### Session scope vs. stored scope

Sometimes, you might want to change which projects are included in your results as you work. Other times, you want to set the project scope and persist it with a resource to keep inputs consistent. Session scope and stored scope are how apps make that happen. The behavior follows the app.

Most apps use **session scope**. Session scope is which projects are searched while you work. You set it with the scope selector in the project header. Your selection is preserved as you navigate between apps that support the selector. Starting a new session resets to the space default. Session scope is used by most apps.

Some apps use **stored scope**. Stored scope is saved with a resource and applies every time that resource runs or opens. Some apps, such as [Dashboards](/explore-analyze/dashboards.md), save a snapshot of the scope currently set in the header selector. Others use a separate selector on the create or edit form, such as [{{ml}} {{anomaly-jobs}}](/explore-analyze/machine-learning/anomaly-detection/ml-ad-run-jobs.md). 

> Refer to the [availability table](#cps-availability) for how each app uses scope.

Author: ```suggestion
In apps where you write queries, you can still [override that scope at the query level](#cps-query-overrides).

Refer to the [availability table](#cps-availability) for how each app uses scope.
```

In apps where you write queries, you can still [override that scope at the query level](#cps-query-overrides).

Refer to the [availability table](#cps-availability) for how each app uses scope.

### Include and exclude projects [cps-picker-include-exclude]

Each project row has a toggle switch that includes or excludes it from searches. You can also use the context menu on any project row for quick actions:

* **Include only this project**: Excludes all other projects currently shown in the list.
* **Exclude only this project**: Includes all other projects currently shown in the list except this one.

You can also select all projects currently shown in the list by clicking **Include all visible**.

The scope selector prevents you from excluding every project. The toggle on the last included project is deactivated, so at least one project is always included.

If your current selection results in zero included projects, the scope selector displays a warning.

### Filter projects by tag [cps-picker-tag-filters]

> You can narrow the project list in the scope selector by creating tag filters. Tag filters let you find and select projects based on [predefined and custom project tags](/explore-analyze/cross-project-search/cross-project-search-tags.md). Predefined tags include `_type`, `_region`, and `_csp`. You can also filter on [custom tags](/deploy-manage/deploy/elastic-cloud/project-settings.md#project-tags) that you define in the {{ecloud}} UI.

Author: ```suggestion
You can narrow the project list in the scope selector by creating tag filters. Tag filters let you find and select projects based on [predefined and custom project tags](/explore-analyze/cross-project-search/cross-project-search-tags.md). Predefined tags include `_type`, `_region`, and `_csp`. You can also filter on [custom tags](/deploy-manage/deploy/elastic-cloud/project-settings.md#project-tags) that you define in the {{ecloud}} UI.

:::{note}
When you use a filter by custom tags, [changing the custom tags](/deploy-manage/deploy/elastic-cloud/project-settings.md#project-tags) on a project can include or exclude that project from your selected scope without the scope being edited.
:::
```

5. Select **Apply** ({icon}`check`) to add the filter.

When multiple filters are active, they are combined with AND logic: a project must match all filters to appear in the list.

#### Manage tag filters

Active filters appear as badges below the filter form. You can:

* Select a filter badge to edit its tag, operator, or value.
* Invert a filter to switch between include and exclude logic.
* Turn off a filter without removing it.
* Remove a filter to delete it.

To remove all active filters at once, select **Clear project tag filters** in the scope selector.

#### Quick-filter from project tags

You can create a filter directly from a project's tags. Select the tag count badge on a project row to open a popover listing that project's tags, then select a tag to create a filter for it.

### How newly linked projects affect your scope [cps-picker-new-projects]

Your scope is saved as a [project routing expression](/explore-analyze/cross-project-search/cross-project-search-project-routing.md), not a fixed list, so it also applies to projects that are linked later:

* Projects that match your active tag filters are included automatically.

> * Excluding a project individually affects only that project.

Reviewer: is it worth highlight that this only happens for the space configuration, in other parts of Kibana teams have opted for the fixed list approach e.g. transforms, ml

To keep future projects out of your scope, use a tag filter rather than excluding projects one by one.

:::{note}

> When you use a tag filter, , changing a project's [custom tags](/deploy-manage/deploy/elastic-cloud/project-settings.md#project-tags) can include or exclude that project from your selected scope without the scope being edited.

Author: ```suggestion
When you use a tag filter, changing a project's [custom tags](/deploy-manage/deploy/elastic-cloud/project-settings.md#project-tags) can include or exclude that project from your selected scope without the scope being edited.
```

> :::

Author: ```suggestion
```

:::

To add a tag filter:

1. Open the scope selector and select **Add project tag filter**.
2. Choose a tag from the **Select a tag** dropdown.
3. Choose an operator. The default operator is **is**.
4. If the operator requires a value, choose one or more values from the **Select a value** dropdown.
5. Select **Apply** ({icon}`check`) to add the filter.

When multiple filters are active, they are combined with AND logic: a project must match all filters to appear in the list.

#### Manage tag filters

Active filters appear as badges below the filter form. You can:

* Select a filter badge to edit its tag, operator, or value.
* Invert a filter to switch between include and exclude logic.
* Turn off a filter without removing it.
* Remove a filter to delete it.

To remove all active filters at once, select **Clear project tag filters** in the scope selector.

#### Quick-filter from project tags

You can create a filter directly from a project's tags. Select the tag count badge on a project row to open a popover listing that project's tags, then select a tag to create a filter for it.

### How newly linked projects affect your scope [cps-picker-new-projects]

> Your scope is saved as a [project routing expression](/explore-analyze/cross-project-search/cross-project-search-project-routing.md), not a fixed list, so it also applies to projects that are linked later:

Author: ```suggestion
In most cases, your scope is saved as a [project routing expression](/explore-analyze/cross-project-search/cross-project-search-project-routing.md), not a fixed list, so it also applies to projects that are linked later:
```

The **{{cps-cap}} ({{cps-init}}) scope** selector ({icon}`cross_project_search`) in your project's header lets you control which linked projects your searches include.

The scope selector lists all linked projects. The origin project always appears first, labeled **This project**. Each project row shows the project name, project type icon, and any assigned [tags](/explore-analyze/cross-project-search/cross-project-search-tags.md).

From the scope selector, you can:

* [Toggle individual projects on or off](#cps-picker-include-exclude) to include or exclude them from searches.
* [Filter the project list by tags](#cps-picker-tag-filters) to find and select projects based on metadata or custom tags.

The footer displays the count of included and excluded projects.

The scope selector is not editable in every app. Some apps display it as **read-only**, meaning the app uses the space default scope but you cannot change it. In read-only mode, filter badges are visible but you cannot create, edit, or remove filters. Other apps show it as **unavailable**, meaning the app searches only the current project. Refer to [{{cps-cap}} availability by app](#cps-availability) for details.

> When the current selection matches the [space default](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md#cps-default-search-scope), the scope selector displays a **Using space defaults** indicator. When you change the selection, the indicator disappears. To restore the space default at any time, select **Revert to space default** in the scope selector.

Reviewer 2: to check depending on https://github.com/elastic/kibana/pull/279522 if it'll actually be the case or only there for pages forcing the space defaults

The scope selector also provides shortcuts to admin settings. Select **Adjust space defaults** to open the space's {{cps-init}} scope configuration, or **Manage cross-project search** to open the [{{cps-init}} management page](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md).

### Session scope vs. stored scope

> In most apps, the scope selector controls the **session scope**: which projects are searched while you work. Your selection is preserved as you navigate between apps, and starting a new session resets to the space default.

Reviewer 2: not sure if it's worth noting here that this doesn't apply when going to apps where the scope selector is read-only or unavailable (what's explained line 35)

In some apps, the scope selector sets a **stored scope** on a specific resource instead of the session. For example, in {{ml-app}} Anomaly Detection, the scope selector defines the `project_routing` value saved with each {{anomaly-job}}. Refer to the [availability table](#cps-availability) for details on how each app uses the scope selector.

### Include and exclude projects [cps-picker-include-exclude]

Each project row has a toggle switch that includes or excludes it from searches. You can also use the context menu on any project row for quick actions:

* **Include only this project**: Excludes all other projects currently shown in the list.
* **Exclude only this project**: Includes all other projects currently shown in the list except this one.

> You can also select all project currently shown in the list by clicking **Include all listed**.

Reviewer 2: ```suggestion
You can also select all projects currently shown in the list by clicking **Include all listed**.
```

---

# Managing {{cps}} scope in your project apps

When [{{cps}} ({{cps-init}})](/explore-analyze/cross-project-search.md) is enabled and projects are linked, searches initiated from your project's apps run across all linked projects by default. {{kib}} provides several ways to narrow or change this scope:

* **Space default**: Admins [configure a default scope for each space](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md#cps-default-search-scope), which applies when you start a new session.
* **Session scope**: Use the [{{cps-init}} scope selector](#cps-in-kibana) in the project's header to change which projects are searched during your session.
* **Stored scope**: Some features use the [scope selector](#cps-in-kibana) to set a project routing value that is saved with a specific resource, such as a dashboard, an {{anomaly-job}}, or a transform. The stored scope applies every time that resource runs or opens, independent of the session scope.
* **Query-level override**: Use project routing or qualified index expressions in individual queries to target specific projects.

## {{cps-cap}} scope selector [cps-in-kibana]

> The **{{cps-cap}} ({{cps-init}}) scope** selector ({icon}`cross_project_search`) in your project's header lets you control which linked projects your searches include.

Author: ```suggestion
The **{{cps-cap}} ({{cps-init}}) scope** selector ({icon}`cross_project_search`) lets you choose which linked projects to include. It appears in the project header, or on the resource you are creating or editing.
```

When the current selection matches the [space default](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md#cps-default-search-scope), the scope selector displays a **Using space defaults** indicator. When you change the selection, the indicator disappears. To restore the space default at any time, select **Revert to space defaults** in the scope selector.

The scope selector also provides shortcuts to admin settings. Select **Adjust space defaults** to open the space's {{cps-init}} scope configuration, or **Manage cross-project search** to open the [{{cps-init}} management page](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md).

### Session scope vs. stored scope

In most apps, the scope selector controls the **session scope**: which projects are searched while you work. Your selection is preserved as you navigate between apps that support the scope selector, and starting a new session resets to the space default.

> In some apps, the scope selector sets a **stored scope** on a specific resource instead of the session. For example, in {{ml-app}} Anomaly Detection, the scope selector defines the `project_routing` value saved with each {{anomaly-job}}. Refer to the [availability table](#cps-availability) for details on how each app uses the scope selector.

Author: ```suggestion
### Session scope vs. stored scope

Whether you set session scope or stored scope depends on the app.

**Session scope** is which projects are searched while you work. You set it with the scope selector in the project header. Your selection is preserved as you navigate between apps that support the selector. Starting a new session resets to the space default.

**Stored scope** is saved with a resource and applies every time that resource runs or opens. Some apps, such as [Dashboards](/explore-analyze/dashboards.md), save a snapshot of the scope currently set in the header selector. Others use a separate selector on the create or edit form, such as [{{ml}} {{anomaly-jobs}}](/explore-analyze/machine-learning/anomaly-detection/ml-ad-run-jobs.md). 

Refer to the [availability table](#cps-availability) for how each app uses scope.
```

The scope selector lists all linked projects, narrowed by any active [tag filters](#cps-picker-tag-filters). The origin project always appears first, labeled **This project**. Each project row shows the project name, project type icon, and any assigned [tags](/explore-analyze/cross-project-search/cross-project-search-tags.md).

From the scope selector, you can:

* [Toggle individual projects on or off](#cps-picker-include-exclude) to include or exclude them from searches.
* [Filter the project list by tags](#cps-picker-tag-filters) to find and select projects based on metadata or custom tags.

The footer displays the count of included and excluded projects.

The scope selector is not editable in every app. Some apps display it as **read-only**, meaning the app uses the space default scope but you cannot change it. In read-only mode, filter badges are visible but you cannot create, edit, or remove filters. Other apps show it as **unavailable**, meaning the app searches only the current project. Refer to [{{cps-cap}} availability by app](#cps-availability) for details.

> When the current selection matches the [space default](/deploy-manage/cross-project-search-config/cps-config-access-and-scope.md#cps-default-search-scope), the scope selector displays a **Using space defaults** indicator. When you change the selection, the indicator disappears. To restore the space default at any time, select **Revert to space defaults** in the scope selector.

Reviewer 3: can we add a mention that changing the project selector does not change the space default?

### Session scope vs. stored scope

Sometimes, you might want to change which projects are included in your results as you work. Other times, you want to set the project scope and persist it with a resource to keep inputs consistent. Session scope and stored scope are how apps make that happen. The behavior follows the app.

Most apps use **session scope**. Session scope is which projects are searched while you work. You set it with the scope selector in the project header. Your selection is preserved as you navigate between apps that support the selector. Starting a new session resets to the space default. Session scope is used by most apps.

Some apps use **stored scope**. Stored scope is saved with a resource and applies every time that resource runs or opens. Some apps, such as [Dashboards](/explore-analyze/dashboards.md), save a snapshot of the scope currently set in the header selector. Others use a separate selector on the create or edit form, such as [{{ml}} {{anomaly-jobs}}](/explore-analyze/machine-learning/anomaly-detection/ml-ad-run-jobs.md). 

> 

Reviewer 3: we should add that Project routing in a search query, and qualified index expressions takes precedence over the project selector.  Aka. if the project selector is Local only, I open a discover session and write a ES|QL query with SET project_routing = _alias:projectb,   this overrides the project selector

When multiple filters are active, they are combined with AND logic: a project must match all filters to appear in the list.

#### Manage tag filters

Active filters appear as badges below the filter form. You can:

* Select a filter badge to edit its tag, operator, or value.
* Invert a filter to switch between include and exclude logic.
* Turn off a filter without removing it.
* Remove a filter to delete it.

To remove all active filters at once, select **Clear project tag filters** in the scope selector.

#### Quick-filter from project tags

You can create a filter directly from a project's tags. Select the tag count badge on a project row to open a popover listing that project's tags, then select a tag to create a filter for it.

### How newly linked projects affect your scope [cps-picker-new-projects]

Your scope is saved as a [project routing expression](/explore-analyze/cross-project-search/cross-project-search-project-routing.md), not a fixed list, so it also applies to projects that are linked later:

* Projects that match your active tag filters are included automatically.
* Excluding a project individually affects only that project.

> To keep future projects out of your scope, use a tag filter rather than excluding projects one by one.

Author: ```suggestion
To keep future projects out of your scope, use a tag filter rather than excluding projects one by one.

:::{note}
When you use a tag filter, , changing a project's [custom tags](/deploy-manage/deploy/elastic-cloud/project-settings.md#project-tags) can include or exclude that project from your selected scope without the scope being edited.
:::
```
